### PR TITLE
Remove private `build_span_events`

### PR DESCRIPTION
**Pull request by TonyCTHsu**
**What does this PR do?**

Streamline the method calling by removing private `build_span_events`

**Motivation:**

This private method `build_span_events` is only invoked without any arguments

### CODE REVIEW

**Review by delner — COMMENTED** [review]
What's the reason for this?

I can see `build_span_events` as redundant, but this does more than that. It eliminates the use of `resolve_tags` and options for `**options` over named arguments which are known to have high performance cost.

**Comment by TonyCTHsu** [comment]
1. `resolve_tags` is not eliminated, https://github.com/DataDog/dd-trace-rb/pull/3504/files#diff-fb264450c82590f2bf44f09afc63b6a01bd80e4916333f7c19441240694412a1R384

2. I can remove the double splat usage for `option`.

File: lib/datadog/tracing/tracer.rb [hunk 1]
```diff
@@ -376,48 +376,27 @@ def start_span(
...
-            start_time: start_time,
-            tags: resolve_tags(tags),
-            type: type
-          )
-
-          span.start(start_time)
-          span
+          trace.build_span(name, start_time: start_time, **options).tap do |span|
+            span.start(start_time)
+          end
```
Comment (marcotc) [inline lib/datadog/tracing/tracer.rb:395]: Using `**kwargs` it's slower than passing all the arguments.

Also, using `tap` is slower than calling the method directly, then returning the original value.

Comment (marcotc) [reply lib/datadog/tracing/tracer.rb:395]: .I recommend passing the values by name, similar to how they were before, and not using `tap`.

Comment (marcotc) [reply lib/datadog/tracing/tracer.rb:395]: Don't take my word for it, our microbenchmarks should tell us the real numbers: `benchmarks/tracing_trace.rb`

**Review by marcotc — APPROVED** [review]
No externally facing changes. 👍

**Review by delner — APPROVED** [review]

**Pull request merged**